oak_functions/loader/src/server.rs

```diff
@@ -777,24 +779,36 @@ mod tests {
     };
     use maplit::hashmap;
 
-    fn create_test_wasm_handler() -> WasmHandler {
-        let logger = Logger::for_test();
-        let lookup_data = Arc::new(LookupData::for_test(hashmap! {}));
-        let lookup_factory =
-            LookupFactory::new_boxed_extension_factory(lookup_data, logger.clone())
-                .expect("could not create LookupFactory");
+    #[test]
+    fn test_start_from_empty_endpoints() {
+        fn check_empty(endpoint: &mut Endpoint) {
+            let receiver = &mut endpoint.receiver;
+            assert_eq!(TryRecvError::Empty, receiver.try_recv().unwrap_err());
+        }
+        let (mut module, mut runtime) = channel_create();
+        check_empty(&mut module);
+        check_empty(&mut runtime);
+    }
 
-        let wasm_module_bytes = test_utils::create_some_wasm_module_bytes();
+    #[tokio::test]
+    async fn test_crossed_write_read() {
+        async fn check_crossed_write_read(endpoint1: &mut Endpoint, endpoint2: &mut Endpoint) {
+            let message = String::from("Message").into_bytes();
+            let sender = &endpoint1.sender;
+            let send_result = sender.send(message.clone()).await;
+            assert!(send_result.is_ok());
 
-        create_wasm_handler(&wasm_module_bytes, vec![lookup_factory], logger)
-            .expect("could not create wasm_handler")
-    }
+            let receiver = &mut endpoint2.receiver;
+            let received_message = receiver.recv().await.unwrap();
 
-    fn create_test_wasm_state() -> WasmState {
-        let wasm_handler = create_test_wasm_handler();
-        wasm_handler
-            .init(b"".to_vec())
-            .expect("could not create wasm_state")
+            assert_eq!(message, received_message);
+        }
+
+        let (mut module, mut runtime) = channel_create();
```

Comment:
I would not even mention module and runtime here, just endpoint_0 and endpoint_1 perhaps, or something else more abstract?